spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -227,8 +234,21 @@ public void setCharset(String charset) {
 	 */
 	public void setTemporaryFileSuffix(String temporaryFileSuffix) {
 		Assert.notNull(temporaryFileSuffix, "'temporaryFileSuffix' must not be null");
-		this.hasExplicitlySetSuffix = true;
-		this.temporaryFileSuffix = temporaryFileSuffix;
+		this.hasTempFileExpression = true;
+		this.temporaryFileNameHelper = FileNameHelper.defaultForSuffix(temporaryFileSuffix);
+	}
+
+	/**
+	 * Set the temporary suffix to use when transferring files to the remote system.
+	 * Default ".writing".
+	 * @param temporaryFileNameExpression the suffix
+	 * @see #setUseTemporaryFileName(boolean)
+	 */
+	public void setTemporaryFileNameExpression(String temporaryFileNameExpression) {
+		Assert.isTrue(StringUtils.hasText(temporaryFileNameExpression),
+				"'temporaryFileNameExpression' must have a value");
+		this.hasTempFileExpression = true;
+		this.temporaryFileNameHelper = new FileNameHelper("#root" + temporaryFileNameExpression);
```

Comment:
It would blow up.

This is why I added the static methods on the helper (and makes me want to consider injecting the helper rather than the expression).